app/models/result_link.rb

```diff
@@ -0,0 +1,9 @@
+# frozen_string_literal: true
+
+class ResultLink < ApplicationRecord
+  belongs_to :keyword, inverse_of: :result_links
+
+  enum link_type: { ads_top: 0, ads_page: 1, non_ads: 2 }
+
+  validates :url, presence: true, allow_blank: false
```

Comment:
why have we added `allow_blank`: false? 🤔

Reply:
My bad, I was confused between validators and options to omit validation. 
If I understand well, `allow_blank` is an option meaning that by default it's set to false (validation occurs blank or not blank) but once set to true, validation will not be triggered if the :url is blank. 
Thanks for pointing that out @ankitkalia1195 !! :)